upgrade/boot_loader/VerifierEcDsa.cpp

```diff
@@ -1,29 +1,66 @@
 #include "upgrade/boot_loader/VerifierEcDsa.hpp"
+#include "infra/util/ByteRange.hpp"
+#include "infra/util/MemoryRange.hpp"
 #include "mbedtls/sha256.h"
 #include "uECC.h"
 
 namespace application
 {
     VerifierEcDsa::VerifierEcDsa(infra::ConstByteRange key)
         : key(key)
-    {
-        assert(key.size() == 56);
-    }
+    {}
 
     bool VerifierEcDsa::IsValid(hal::SynchronousFlash& flash, const hal::SynchronousFlash::Range& signature, const hal::SynchronousFlash::Range& data) const
     {
         std::array<uint8_t, 32> messageHash = Hash(flash, data);
-        std::array<uint8_t, 56> storedSignature;
+        auto storedSignature = GetSignatureStorage(signature);
 
-        if (signature.second - signature.first != storedSignature.size())
+        if (storedSignature.empty())
             return false;
 
-        flash.ReadBuffer(storedSignature, signature.first);
-
+        flash.ReadBuffer(infra::ConstCastByteRange(storedSignature), signature.first);
         return uECC_verify(key.begin(),
                    messageHash.data(),
                    messageHash.size(),
-                   storedSignature.data(),
-                   uECC_secp224r1()) == 1;
+                   storedSignature.begin(),
+                   GetCurve()) == 1;
+    }
+
+    VerifierEcDsa224::VerifierEcDsa224(infra::ConstByteRange key)
+        : VerifierEcDsa(key)
+    {
+        assert(key.size() == signSize);
+    }
+
+    infra::ConstByteRange VerifierEcDsa224::GetSignatureStorage(const hal::SynchronousFlash::Range& signature) const
+    {
+        if (signature.second - signature.first != signatureStorage.size())
+            return infra::ConstByteRange();
+        else
+            return signatureStorage;
+    }
```

Comment:
The intention for the value returned is to be written to, so it should be modifiable

```suggestion
    infra::ByteRange VerifierEcDsa224::GetSignatureStorage(const hal::SynchronousFlash::Range& signature) const
    {
        if (signature.second - signature.first != signatureStorage.size())
            return infra::ByteRange();
        else
            return signatureStorage;
    }
```